src/WebJobs.Script.WebHost/Security/KeyManagement/SimpleKubernetesClient.cs

```diff
@@ -0,0 +1,307 @@
+                if (response.IsSuccessStatusCode)
+                {
+                    var obj = await response.Content.ReadAsAsync<JObject>();
+                    return obj["data"]
+                        ?.ToObject<IDictionary<string, string>>()
+                        ?.ToDictionary(
+                            k => k.Key,
+                            v => decode
+                                ? Encoding.UTF8.GetString(Convert.FromBase64String(v.Value))
+                                : v.Value);
+                }
+                else if (response.StatusCode == HttpStatusCode.NotFound)
+                {
+                    // If the collection is not there, return an empty list
+                    return new Dictionary<string, string>();
+                }
+                else
+                {
+                    throw new HttpRequestException($"Error calling GET {url}, Status: {response.StatusCode}, Content: {await response.Content.ReadAsStringAsync()}");
+                }
+            }
+        }
+
+        private async Task CreateIfDoesntExist(string url, bool isSecret)
+        {
+            using (var request = await GetRequest(HttpMethod.Get, url))
+            {
+                var response = await _httpClient.SendAsync(request);
+                if (response.StatusCode == HttpStatusCode.NotFound)
+                {
+                    var name = url.Split("/").Last();
+                    url = url.Substring(0, url.LastIndexOf("/"));
+                    var payload = new
+                    {
+                        kind = isSecret ? "Secret" : "ConfigMap",
+                        apiVersion = "v1",
+                        metadata = new
+                        {
+                            name = name
+                        }
+                    };
+                    using (var createRequest = await GetRequest(HttpMethod.Post, url, payload))
+                    {
+                        var createResponse = await _httpClient.SendAsync(createRequest);
+                        createResponse.EnsureSuccessStatusCode();
+                    }
+                }
+            }
+        }
+
+        private async Task<(string, bool)> GetObjectUrl(string objectName, bool watchUrl = false)
+        {
+            var isSecret = true;
+            if (!objectName.StartsWith("secrets/") && !objectName.StartsWith("configmaps/"))
+            {
+                // assume a secret if no type is specified.
+                objectName = $"secrets/{objectName}";
+            }
+            else if (objectName.StartsWith("configmaps/"))
+            {
+                // This allows the users to manage their keys in configMaps instead of secret.
+                // There is no difference currently in kubernetes between the 2 other than secrets
+                // requiring base64 encoding. If it's a configMap, then we don't need to decode it.
+                isSecret = false;
+            }
```

Comment:
We assume a `Secret` but if the user sets `configmaps/` then assume a `ConfigMap` instead.